docs/book/how-to/customize-docker-builds/use-a-prebuilt-image.md

```diff
@@ -0,0 +1,122 @@
+---
+description: "Skip building an image for your ZenML pipeline altogether."
+---
+
+# Use a prebuilt image for pipeline execution
+
+When running a pipeline on a remote Stack, ZenML builds a Docker image with a base ZenML image and adds all of your project dependencies and your pipeline code to it. This process might take significant time depending on how big your dependencies are, how powerful your local system is and how fast your internet connection is (to pull base layers and push the final image to your container registry). Although this process only happens once and is skipped if ZenML detects no change in your environment, it might still be a bottleneck in your pipeline execution.
+
+To save time and costs, you can choose to not build a Docker image every time your pipeline runs. This guide shows you how to do it using a pre-built image, what you should include in your image for the pipeline to run successfully and other tips.
+
+{% hint style="info" %}
+Note that using this feature means that you won't be able to leverage any updates you make to your code or dependencies, outside of what your image already contains.
+{% endhint %}
+
+The DockerSettings class in ZenML allows you to set a parent image to be used in your pipeline runs and the ability to skip building an image on top of it. This approach works in two ways:
+- [If ZenML built an image for your pipeline previously](#if-zenml-built-an-image-for-your-pipeline-previously)
+- [If ZenML has never built an image for your pipeline](#if-zenml-has-never-built-an-image-for-your-pipeline)
+
+## If ZenML built an image for your pipeline previously
+
+This is the case where you have had a previous pipeline run where ZenML did build an image and pushed it your registry.
+
+Here, you can just reuse the image by setting it to the `parent_image` attribute of the `DockerSettings` class and setting `skip_build` to `True`.
+
+```python
+docker_settings = DockerSettings(
+    parent_image="my_registry.io/image_name:tag",
+    skip_build=True
+)
+
+
+@pipeline(settings={"docker": docker_settings})
+def my_pipeline(...):
+    ...
+```
+
+On every subsequent pipeline run now, ZenML will just use the code and the dependencies that were a part of the pipeline run whose image you are now using. This approach works without you having to worry about what goes in the image since ZenML built it in the first place.
```

Comment:
```suggestion
With this configuration, every subsequent pipeline run now will have ZenML use the code and the dependencies that were a part of the pipeline run whose image you specified. This approach works without you having to worry about what goes in the image since ZenML built it in the first place.
```